eNMS/services/file_transfer/generic_file_transfer.py

```diff
@@ -85,15 +85,19 @@ def job(self, run, payload, device):
                     device,
                 )
                 run.transfer_file(ssh_client, pairs)
-        elif not Path(source).is_file():
+        elif run.direction.lower() == "put" and not Path(source).is_file():
             success = False
             result = (
                 f"Source file {source} does not exist, is a directory,"
                 " or you forgot to enable globbing"
             )
         else:
             run.log("info", f"Transferring file {source} to {destination}", device)
-            run.transfer_file(ssh_client, [(source, destination)])
+            try:
+                run.transfer_file(ssh_client, [(source, destination)])
+            except FileNotFoundError:
+                success = False
+                result = f"Source file {source} does not exist, or you forgot to enable globbing"
```

Comment:
there's duplicated code here. If we're going to catch exception when transferring the file, why bother having this `elif` check ?

Reply:
Agreed